etna/transforms/decomposition/base_change_points.py

```diff
@@ -0,0 +1,104 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import List
+from typing import Tuple
+from typing import Type
+
+import pandas as pd
+from ruptures.base import BaseEstimator
+from ruptures.costs import CostLinear
+from sklearn.base import RegressorMixin
+
+TTimestampInterval = Tuple[pd.Timestamp, pd.Timestamp]
+TDetrendModel = Type[RegressorMixin]
+
+
+class BaseChangePointsModelAdapter(ABC):
+    """BaseChangePointsModelAdapter is the base class for change point models adapters."""
+
+    @abstractmethod
+    def get_change_points(self, series: pd.Series) -> List[pd.Timestamp]:
+        """Find change points within one segment.
+
+        Parameters
+        ----------
+        series:
+            series in which to look for change points
+
+        Returns
+        -------
+        change points:
+            change point timestamps
+        """
+        pass
+
+    @staticmethod
+    def _build_intervals(change_points: List[pd.Timestamp]) -> List[TTimestampInterval]:
+        """Create list of stable intervals from list of change points."""
+        change_points.extend([pd.Timestamp.min, pd.Timestamp.max])
+        change_points = sorted(change_points)
+        intervals = list(zip(change_points[:-1], change_points[1:]))
+        return intervals
+
+    def get_change_points_intervals(self, df: pd.DataFrame, in_column: str) -> List[TTimestampInterval]:
+        """Find change point intervals in given dataframe and column.
+
+        Parameters
+        ----------
+        df:
+            dataframe indexed with timestamp
+        in_column:
+            name of column to get change points
+
+        Returns
+        -------
+        :
+            change points intervals
+        """
+        series = df.loc[df[in_column].first_valid_index() : df[in_column].last_valid_index(), in_column]
```

Comment:
I guess this logic is not common for all the change point models, so it should be in the implementation of `get_change_points`. Change the signature of the `get_change_points(self, df: pd.Dataframe)`